src/test/java/com/blibli/future/controller/ShiftControllerTest.java

```diff
@@ -0,0 +1,175 @@
+package com.blibli.future.controller;
+
+import com.blibli.future.exception.IdNotFoundException;
+import com.blibli.future.model.Shift;
+import com.blibli.future.service.api.ShiftService;
+import com.blibli.future.valueObject.ShiftVO;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import net.minidev.json.JSONArray;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ShiftControllerTest {
+    @InjectMocks
+    private ShiftController shiftController;
+    @Mock
+    private ShiftService shiftService;
+
+    private MockMvc mockMvc;
+
+    private final String BASE_PATH = "/shift";
+    private final String PATH_WITH_ID = BASE_PATH + "/test";
```

Comment:
My "comparative study" said that we don't need to write this. Just use the one declared as public in the Controller